File: in_common/src/main/java/it/unical/ea/dtos/review/ReviewDto.java
package it.unical.ea.dtos.review;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

import java.time.LocalDateTime;
import java.util.UUID;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class ReviewDto {
    private UUID id;
    private String authorName;
    private Boolean isEditable;
    private Double rating;
    private String comment;
    private LocalDateTime createdAt;
    
    private UUID activityId;
    private String activityName; // Populated when returning aggregated reviews for an itinerary

    private UUID itineraryId;
    private String itineraryName;
}

File: frontend/app/src/main/java/com/travel/app/domain/model/review/ReviewDto.kt
package com.travel.app.domain.model.review

data class ReviewDto(
    val id: String? = null,
    val authorName: String,
    val isEditable: Boolean = false,
    val rating: Double,
    val comment: String? = null,
    val createdAt: String? = null,
    val activityId: String? = null,
    val activityName: String? = null,
    val itineraryId: String? = null,
    val itineraryName: String? = null
)

data class CreateReviewDto(
    val activityId: String? = null,
    val itineraryId: String? = null,
    val rating: Double,
    val comment: String? = null
)
